src/adder.rs

```diff
@@ -1,58 +1,52 @@
 use anyhow::{Context, Result};
-use rnix::SyntaxNode;
+use rnix::{SyntaxNode};
+
+use crate::verify_getter::SyntaxNodeAndWhitespace;
 
 pub fn add_dep(
-    contents: &mut String,
-    deps_list: SyntaxNode,
+    deps_list: SyntaxNodeAndWhitespace,
     new_dep_opt: Option<String>,
-) -> Result<String> {
+) -> Result<SyntaxNode> {
+
     let new_dep = new_dep_opt.context("error: no dependency")?;
+    let whitespace = deps_list.whitespace;
+    let deps_list = deps_list.node;
 
     for dep in deps_list.children() {
         if dep.to_string() == new_dep {
             // dep is already present in the deps_list, we're done
-            return Ok(contents.to_string());
+            return Ok(deps_list);
         }
     }
 
-    let dep_list_first_token = deps_list
-        .first_token()
-        .context("error: could not find first bracket token in deps list")?;
-    let open_bracket_pos: usize = dep_list_first_token.text_range().start().into();
+    let mut base_indent = 0;
+    if let Some(w) = whitespace {
+        base_indent = w.text().replace("\n", "").len();
+    }
+    let entry_indent = base_indent + 2;
+    let base_whitespace = " ".repeat(base_indent);
+    let newline_whitespace = &format!("\n{base_whitespace}");
 
-    // add dep pos is the character position of the first character of the new dependency
-    let add_dep_pos = calc_add_dep_pos(deps_list);
+    let has_newline = deps_list
+        .to_string()
+        .contains('\n');
 
-    // we need to add leading whitespace to the next line so that
-    // the pkgs are correctly formatted (i.e. they are lined up)
-    let white_space_count = if add_dep_pos >= 2 + open_bracket_pos {
-        add_dep_pos - open_bracket_pos - 2
-    } else {
-        2
+    let newline =  match has_newline {
+        true => "",
+        false => newline_whitespace,
```

Comment:
can you check for the newline first and then determine if you need `\n` at the beginning of the string?

Reply:
oh you don't need `base_whitespace` later either? instead can you put all the string construction logic inside of the `false` match arm?